### ISSUE TITLE
How to access this site, totally decentralized (like ZeroNet)?

### ISSUE THREAD

**Issue by DaniellMesquita** [ipfs-search/ipfs-search#100]
Can this site be accessed through only IPFS?
And peers running server actions (NodeJS) and Database in a p2p way?

**Comment by dokterbob** [comment]
Ref: https://docs.ipfs.io/reference/cli/#ipfs-p2p-listen

**Comment by DaniellMesquita** [comment]
@dokterbob
> a) Funding (i.e. grants - we might get them from Protocol Labs AFAIK).

Suggestion: Gitcoin

**Comment by dokterbob** [comment]
@DaniellMesquita We've actually made significant steps with funding acquisition (although, yes, Gitcoin is in our view). That is to say, we have funding available and are preparing to put bounties on tasks.

We have identified 2 workstreams we'd like to explore:

1. Offer tunnelled services through the IPFS overlay network to provide censorship-resistant access to our search engine.
2. A decentralized search prototype akin to [TorrentParadise](https://torrent-paradise.ml/) 

If you (or anyone) are seriously interested and able to take on the implementation, please reach out to us.

**Comment by dokterbob** [comment]
@urbanguacamole Would you perhaps be interested in getting paid to make a PoC for decentralized search over (part of) our index?

**Comment by urbanguacamole** [comment]
@dokterbob sorry for the late reply, I've been busy the past month. i'm actually not the author of the backend ipfsearch stuff, i only use parts of github.com/tensojka/ipfsearch-webapp (or the indexer). i talked to the author (@tensojka), he helps me with debugging sometimes. I think it would be better to ask him directly.

**Comment by dokterbob** [comment]
Hi @urbanguacamole, better late than never! We've all got busy lives, don't worry.

@tensojka @urbanguacamole I think we've had a video call one day. I have been working with @askender to get a JSON export of just titles. Would you both or one of you be interested in trying to index the titles of documents in our index with your decentralized index? We have funding available (as long as it's Open Source).

**Comment by askender** [comment]
@dokterbob   My open-source and open-data Dweb search engine is in alpha version now:  https://dweb-search.on.fleek.co/   ipns://dweb-search.on.fleek.co/    The frontend and the exported jsonl snpashot is in  ipfs/ipns. The backend only rely on python and mongodb, everyone can fork and host it,  will be more decentralized too.    

> JSON export of just titles
Sounds similar with my comment: https://github.com/ipfs-search/ipfs-search/issues/190#issuecomment-878896718
{'hash': 'Qm...', 'title': '***'}
{'hash': 'Qm...', 'title': '***'}

> Trying to index the titles of documents with my decentralized index
Of cource I will do that.  I don't add ElasticSearch in my code (maybe I will make it optional in the future), so I will get keywords from title and html-metadata to add it in the json and mongodb.

Besides, I will try to find ways to get as much ipns blogs/sites as possible.

**Comment by dokterbob** [comment]
@askender Yes, it's based on your proposal. Maybe we could make this a little collaboration?
We'd love to share our data for proper decentralised search :)

I'd love to see to what size the 'essence' of our data has. Would be good to have references (filenames) too, I guess.

**Comment by dokterbob** [comment]
We'd love to crawl IPNS too. It shouldn't be too hard but there's already so much going on.

**Comment by dokterbob** [comment]
@madnificent would like to get involved here and have a chat. 

Could we invite you all to join us on Matrix? https://matrix.to/#/#ipfs-search:matrix.org

@askender @urbanguacamole @danimesq @tensojka
